### PR TITLE
Replace NPM with Yarn

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?

As per [this Trello card](https://trello.com/c/DdZ0APNI/678-3-fix-versions-of-node-modules), this PR replaces NPM with [Yarn](https://yarnpkg.com/).

#### Bonus Content

Also cleans up some dependencies and removes unused pattern library css.

### How to review
1. Remove `node_modules` directory.
2. `yarn` to install dependencies
3. `yarn compile` to compile static assets.
4. `yarn test` to run all front-end tests.

### CODE REVIEW

**Comment by benjefferies** [comment]
Are the css files supposed to be added?

**Comment by hamishtaplin** [comment]
@benjefferies No. There's something weird going on related to moving the `static` assets path.

File: Dockerfile [hunk 1]
```diff
@@ -12,18 +12,18 @@ RUN apt-get update && apt-get install -y $RUNTIME_PACKAGES $BUILD_PACKAGES \
 	&& curl -sL https://deb.nodesource.com/setup_5.x | bash -
 
 RUN apt-get install -y nodejs
-RUN npm install --global gulp-cli
```
Comment (iwootten) [inline Dockerfile:15]: Doesn't look like yarn has actually been added as a dependency here. Guess we need to add the same packages to sources list and apt-install as per the travis file?

File: README.md [hunk 2]
```diff
@@ -67,20 +67,20 @@ npm install
...
-`npm run dev`                              | Build assets and watch for changes. Runs Browsersync.
-`npm run test`                             | Runs the unit tests through Karma and the functional tests through a local Selenium instance
-`npm run test_unit`                        | Watches the unit tests via Karma
-`npm run test_functional`                  | Runs the functional tests through a local Selenium instance (requires app running on localhost:5000)
-`npm run test_functional_sauce`            | Runs the functional tests through Sauce Labs (requires app running on localhost:5000)
-`npm run lint`                             | Lints the JS, reporting errors/warnings.
+`yarn compile`                          | Build the assets (js, css, img) into `/static`
+`yarn dev`                              | Build assets and watch for changes. Runs Browsersync.
+`yarn test`                             | Runs the unit tests through Karma and the functional tests through a local Selenium instance
+`yarn test_unit`                        | Watches the unit tests via Karma
```
Comment (benjefferies) [inline README.md:80]: Doubt it's related to the change to yarn but `yarn test_unit` hangs after showing the summary

Comment (hamishtaplin) [reply README.md:80]: @benjefferies That is a problem but wasn't introduced by this PR.

**Review by benjefferies — APPROVED** [review]

**Pull request merged**